src/twisted/names/authority.py

```diff
@@ -270,141 +289,253 @@ def setupConfigNamespace(self):
         return r
 
 
+
 class BindAuthority(FileAuthority):
-    """An Authority that loads BIND configuration files"""
+    """
+    An Authority that loads U{BIND zone files
+    <https://en.wikipedia.org/wiki/Zone_file>}.
 
+    Supports only C{$ORIGIN} and C{$TTL} directives.
+    """
     def loadFile(self, filename):
-        self.origin = os.path.basename(filename) + '.' # XXX - this might suck
+        """
+        This is a private API; do not use in application code.
 
-        with open(filename, 'rb') as f:
-            lines = f.readlines()
+        Load records from C{filename}.
+
+        @param filename: file to read from
+        @type filename: L{bytes}
+        """
+        fp = FilePath(filename)
+        # XXX - not the best way to set an origin. It can be set using $ORIGIN
+        # though.
+        self.origin = nativeString(fp.basename() + b'.')
+
+        lines = fp.getContent().splitlines(True)
         lines = self.stripComments(lines)
         lines = self.collapseContinuations(lines)
         self.parseLines(lines)
 
 
     def stripComments(self, lines):
-        return [
-            a.find(';') == -1 and a or a[:a.find(';')] for a in [
+        """
+        This is a private API; do not use in application code.
```

Comment:
Can you make this method private and then add a deprecated public alias?  Deprecation should be communicated structurally in a way that automated tooling can deal with, not with ad-hoc doc comments.